opentelemetry-exporter-geneva/geneva-uploader/src/ingestion_service/mod.rs

```diff
@@ -0,0 +1,242 @@
+pub(crate) mod uploader;
+
+#[cfg(test)]
+mod tests {
+    use crate::{AuthMethod, GenevaConfigClient, GenevaConfigClientConfig};
+    use crate::{GenevaUploader, GenevaUploaderConfig};
+
+    use std::env;
+    use std::fs;
+    use std::time::Instant;
+
+    #[tokio::test]
+    /// To run this test against a real Geneva Config Service and GIG, set the following environment variables:
+    ///
+    /// ```bash
+    /// export GENEVA_ENDPOINT="xxxhttps://<gcs-endpoint>"
+    /// export GENEVA_ENVIRONMENT="Test"
+    /// export GENEVA_ACCOUNT="YourAccountName"
+    /// export GENEVA_NAMESPACE="YourNamespace"
+    /// export GENEVA_REGION="YourRegion"
+    /// export GENEVA_CONFIG_MAJOR_VERSION="2"
+    /// export GENEVA_CERT_PATH="/path/to/client.p12"
+    /// export GENEVA_CERT_PASSWORD="your-cert-password"
+    /// export GENEVA_SOURCE_IDENTITY="Tenant=YourTenant/Role=YourRole/RoleInstance=YourInstance"
+    /// export GENEVA_BLOB_PATH="/path/to/blob.bin"
+    ///
+    /// cargo test test_upload_to_gig_real_server -- --ignored --nocapture
+    /// ```
+    #[ignore]
+    async fn test_upload_to_gig_real_server() {
+        // === 1. Load binary blob ===
+        let blob_path = env::var("GENEVA_BLOB_PATH").expect("GENEVA_BLOB_PATH env var is required");
```

Comment:
nit: You could extract the common functionality of reading the environment variables and creating the config client and uploader into a function and reuse them in both the tests.

Reply:
Good point. Done.